src/main.cpp

```diff
@@ -2577,7 +2577,7 @@ bool FindBlockPos(CValidationState &state, CDiskBlockPos &pos, unsigned int nAdd
         pos.nPos = vinfoBlockFile[nFile].nSize;
     }
 
-    if (nFile != nLastBlockFile) {
```

Comment:
I think it could - but that would require having to check whether that breaks anything...and if we're going down that route, we could also start changing a bunch of other variables...and I don't think I want to go down that path right now :)